### PR TITLE
mrc-2313 Build chart metadata

### PR DESCRIPTION
**Pull request by EmmaLRussell**
Sorry, this looks like another massive changeset - most of it is chart config files copied directly from the [proof of concept project](https://github.com/mrc-ide/covidsim-charts-gui).

This branch:
- Adds these files to the app as separate resource files
- Adds an AppMetadata component in the SpringBoot app - this builds a metadata response from these files on application startup
- Adds a metadata controller and endpoint
- Adds action to fetch metadata, related mutation and call to action from the Home component in the front end
- Some development of the store and related testing in the front end:
    - Separate out actions and mutations into their own files
    - Add a RootState type, and very basic typing for the metadata - this can be developed further in other tickets
    - Add some mocking support for front end testing

So if you look at network traffic in browser when you run up the app you'll be able the the metadata request and response. 

We don't have front end integration support yet - that feels like a job for another ticket. Similarly for error handling in front end.

### CODE REVIEW

File: src/app/static/.eslintrc.js [hunk 1]
```diff
@@ -16,15 +16,20 @@ module.exports = {
         "no-debugger": process.env.NODE_ENV === "production" ? "warn" : "off",
         "comma-dangle": ["error", "never"],
         indent: ["error", 4],
-        quotes: ["error", "double", { avoidEscape: true }]
+        quotes: ["error", "double", { avoidEscape: true }],
+        "arrow-body-style": "off",
```
Comment (EmmaLRussell) [inline src/app/static/.eslintrc.js:20]: These two rules don't seem to work well with Typescript - `arrow-body-style` seems to interpret a return type as a function body without braces. `prefer-default-export`. And I don't think you can specify a type with default export, hence removing `prefer-default-export`.

**Comment by EmmaLRussell** [comment]
@mwoodbri I just saw your comment on the ticket about replacing config.json with jsonata - makes sense, happy to do that in this branch or another.

File: src/app/build.gradle.kts [hunk 2]
```diff
@@ -18,6 +18,7 @@ dependencies {
     implementation("com.github.kittinunf.fuel:fuel:2.3.1")
     implementation("org.jetbrains.kotlin:kotlin-reflect")
     implementation("net.logstash.logback:logstash-logback-encoder:6.6")
+    implementation("com.fasterxml.jackson.module:jackson-module-kotlin")
```
Comment (mwoodbri) [inline src/app/build.gradle.kts]: I think you can pin this to `2.12.3` and remove L26.

**Review by mwoodbri — CHANGES_REQUESTED** [review]
I think if we're going to make `config` a string then this is the place to do it. But PR can be merged almost immediately thereafter - looks great and seems like we should nearly be able to render the actual charts with mock data as in the prototype.

Just one comment inline. Have added mrc-2358 for other unrelated things I spotted.

**Review by mwoodbri — CHANGES_REQUESTED** [review]
All looks great... except minor issue not addressed since original review

**Comment by EmmaLRussell** [comment]
> All looks great... except minor issue not addressed since original review

:facepalm: Sorry!

**Review by mwoodbri — APPROVED** [review]

**Pull request merged**